contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,919 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.17;
+
+// Created By: Art Blocks Inc.
+
+import "./interfaces/0.8.x/IAdminACLV0.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "./interfaces/0.8.x/IDependencyRegistryV0.sol";
+
+import "@openzeppelin-4.7/contracts/utils/Strings.sol";
+import "@openzeppelin-4.7/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+/**
+ * @title Art Blocks Dependency Registry, V0.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * Permissions managed by ACL contract
```

Comment:
@yoshiwarab Given that we do not typically use the proxy upgradable pattern at Art Blocks, I think it is worth expanding on the "why" aspect of why we've implemented this contract to allow for upgrades, based on our prior internal conversation on the topic.

Reply:
In doing so–this is also probably a place to call out that if/when we ever do call `renounceOwnership` (which we could do if no longer maintaining the registry, and instead intending to "freeze" it at a fixed period in time) that this becomes a frozen immutable registry, as no upgrades can be performed.

Reply:
Another related thought: we may want to consider testing an upgrade on testnet before deploying to mainnet.

Reply:
@jakerockland, updated the comment here but can't remember exactly what we discussed internally about upgradeability so left it pretty generic. Let me know if you'd like me to add anything else1

Reply:
Added one suggestion here but otherwise LGTM! https://github.com/ArtBlocks/artblocks-contracts/pull/395/files#r1054630021

Thank you ser.